### PR TITLE
[FLINK-20204][docs-zh] Translate page 'upsert-kafka Connector' into Chinese

### PR DESCRIPTION
**Pull request by caozhen1937**
## What is the purpose of the change

Translate page 'upsert-kafka Connector' into Chinese.

## Brief change log

*docs/dev/table/connectors/upsert-kafka.zh.md*
*docs/dev/table/connectors/upsert-kafka.md*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:(no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
cc @fsk119 , could you help to review this?

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 1]
```diff
@@ -101,112 +89,97 @@ GROUP BY region;
...
 
-By default, an Upsert Kafka sink ingests data with at-least-once guarantees into a Kafka topic if
-the query is executed with [checkpointing enabled]({% link dev/stream/state/checkpointing.zh.md %}#enabling-and-configuring-checkpointing).
+默认情况下，如果[启用 checkpoint]({% link dev/stream/state/checkpointing.zh.md %}#enabling-and-configuring-checkpointing)，Upsert Kafka sink 会保证至少一次将数据插入 Kafka topic。
 
-This means, Flink may write duplicate records with the same key into the Kafka topic. But as the
-connector is working in the upsert mode, the last record on the same key will take effect when
-reading back as a source. Therefore, the upsert-kafka connector achieves idempotent writes just like
-the [HBase sink]({{ site.baseurl }}/dev/table/connectors/hbase.html).
+这意味着，Flink 可以将具有相同 key 的重复记录写入 Kafka topic。但由于连接器是在 upsert 模式下工作的，在作为 source 回读时同一 key 的最后一条记录才会生效。因此，upsert-kafka 连接器可以像 [HBase sink]({{ site.baseurl }}/dev/table/connectors/hbase.html) 一样实现幂等写入。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:178]: >但由于连接器是在 upsert 模式下工作的，在作为 source 回读时同一 key 的最后一条记录才会生效。

但由于该连接器以 upsert 的模式工作，该连接器作为 source 读入时，可以确保具有相同主键值下仅最后一条消息会生效。

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 2]
```diff
@@ -29,36 +29,24 @@ under the License.
...
-The Upsert Kafka connector allows for reading data from and writing data into Kafka topics in the upsert fashion.
+Upsert Kafka 连接器支持以 upsert 方式从 Kafka topic 中读取数据并将数据写入 Kafka topic。
 
-As a source, the upsert-kafka connector produces a changelog stream, where each data record represents
-an update or delete event. More precisely, the value in a data record is interpreted as an UPDATE of
-the last value for the same key, if any (if a corresponding key doesn’t exist yet, the update will
-be considered an INSERT). Using the table analogy, a data record in a changelog stream is interpreted
-as an UPSERT aka INSERT/UPDATE because any existing row with the same key is overwritten. Also, null
-values are interpreted in a special way: a record with a null value represents a “DELETE”.
+作为 source，upsert-kafka 连接器生产变更日志流，其中每条数据记录代表一个更新或删除事件。更准确地说，数据记录中的 value 被解释为同一 key 的最后一个 value 的 UPDATE，如果有这个 key（如果不存在相应的 key，则该更新被视为 INSERT）。用表来类比，更改日志流中的数据记录被解释为 UPSERT，也称为 INSERT/UPDATE，因为任何具有相同 key 的现有行都被覆盖。另外，空 value 以特殊方式解释：具有空 value 的记录表示“DELETE”。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:34]: I think we can keep the `changelog` untranslated.  It's more straightforward for users to understand.


>另外，空 value 以特殊方式解释：具有空 value 的记录表示“DELETE”。

另外， value 为空的消息将会被视作为 DELETE 消息

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 3]
```diff
@@ -101,112 +89,97 @@ GROUP BY region;
...
       <td><h5>key.format</h5></td>
-      <td>required</td>
+      <td>必选</td>
       <td style="word-wrap: break-word;">(none)</td>
       <td>String</td>
-      <td>The format used to deserialize and serialize the key part of the Kafka messages. The key part
-      fields are specified by the PRIMARY KEY syntax. The supported formats include <code>'csv'</code>,
-      <code>'json'</code>, <code>'avro'</code>. Please refer to <a href="{% link dev/table/connectors/formats/index.zh.md %}">Formats</a>
-      page for more details and more format options.
+      <td>用于对 Kafka 消息中 key 部分反序列化和序列化的格式。key 字段由 PRIMARY KEY 语法指定。支持的格式包括 <code>'csv'</code>、<code>'json'</code>、<code>'avro'</code>。请参考<a href="{% link dev/table/connectors/formats/index.zh.md %}">格式</a>页面以获取更多详细信息和格式参数。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:134]: >反序列化和序列化

序列化和反序列化?

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 4]
```diff
@@ -101,112 +89,97 @@ GROUP BY region;
...
     <tr>
       <td><h5>value.format</h5></td>
-      <td>required</td>
+      <td>必选</td>
       <td style="word-wrap: break-word;">(none)</td>
       <td>String</td>
-      <td>The format used to deserialize and serialize the value part of the Kafka messages.
-      The supported formats include <code>'csv'</code>, <code>'json'</code>, <code>'avro'</code>.
-      Please refer to <a href="{% link dev/table/connectors/formats/index.zh.md %}">Formats</a> page for more details and more format options.
+      <td>用于对 Kafka 消息中 value 部分反序列化和序列化的格式。支持的格式包括 <code>'csv'</code>、<code>'json'</code>、<code>'avro'</code>。请参考<a href="{% link dev/table/connectors/formats/index.zh.md %}">格式</a>页面以获取更多详细信息和格式参数。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:142]: ditto

File: docs/dev/table/connectors/upsert-kafka.md [hunk 5]
```diff
@@ -47,9 +47,6 @@ key will fall into the same partition.
 Dependencies
 ------------
 
-In order to set up the upsert-kafka connector, the following table provide dependency information for
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.md:50]: Why delete this?

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 6]
```diff
@@ -29,36 +29,24 @@ under the License.
...
-as an UPSERT aka INSERT/UPDATE because any existing row with the same key is overwritten. Also, null
-values are interpreted in a special way: a record with a null value represents a “DELETE”.
+作为 source，upsert-kafka 连接器生产变更日志流，其中每条数据记录代表一个更新或删除事件。更准确地说，数据记录中的 value 被解释为同一 key 的最后一个 value 的 UPDATE，如果有这个 key（如果不存在相应的 key，则该更新被视为 INSERT）。用表来类比，更改日志流中的数据记录被解释为 UPSERT，也称为 INSERT/UPDATE，因为任何具有相同 key 的现有行都被覆盖。另外，空 value 以特殊方式解释：具有空 value 的记录表示“DELETE”。
 
-As a sink, the upsert-kafka connector can consume a changelog stream. It will write INSERT/UPDATE_AFTER
-data as normal Kafka messages value, and write DELETE data as Kafka messages with null values
-(indicate tombstone for the key). Flink will guarantee the message ordering on the primary key by
-partition data on the values of the primary key columns, so the update/deletion messages on the same
-key will fall into the same partition.
+作为 sink，upsert-kafka 连接器可以消费变更日志流。它会将 INSERT/UPDATE_AFTER 数据作为正常的 Kafka 消息写入，并且将 DELETE 数据作为具有空 value 的 Kafka 消息写入(表示 key 对应的 value 被删除)。Flink 将根据主键列的值对数据进行分区，从而保证主键上的消息有序，因此同一 key 上的更新/删除消息将落在同一分区中。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:36]: 变更日志 -> changelog 

> 并且将 DELETE 数据作为具有空 value 的 Kafka 消息写入(表示 key 对应的 value 被删除)

并将 DELETE 数据以 value 为空的的 Kafka 消息写入(表示 对应 key 的消息被删除)

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 7]
```diff
@@ -101,112 +89,97 @@ GROUP BY region;
...
 
-Data Type Mapping
+数据类型映射
 ----------------
 
-Upsert Kafka stores message keys and values as bytes, so Upsert Kafka doesn't have schema or data types.
-The messages are deserialized and serialized by formats, e.g. csv, json, avro. Thus, the data type mapping
-is determined by specific formats. Please refer to [Formats]({% link dev/table/connectors/formats/index.zh.md %})
-pages for more details.
+Upsert Kafka 用字节存储消息的 key 和 value，因此没有 schema 或数据类型。消息按格式进行反序列化和序列化，例如：csv、json、avro。因此数据类型映射表由指定的格式确定。请参考[格式]({% link dev/table/connectors/formats/index.zh.md %})页面以获取更多详细信息。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:183]: 反序列化和序列化 -> 序列化和反序列化

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution @caozhen1937 . I left some comments.

File: docs/dev/table/connectors/upsert-kafka.zh.md (hunk 2 above)
Comment (caozhen1937) [reply docs/dev/table/connectors/upsert-kafka.zh.md:34]: I agree with you.  Keeping the `changelog`.

File: docs/dev/table/connectors/upsert-kafka.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (caozhen1937) [reply docs/dev/table/connectors/upsert-kafka.md:50]: I see FLINK-20093 , so keep the same as it.

File: docs/dev/table/connectors/upsert-kafka.zh.md (hunk 7 above)
Comment (caozhen1937) [reply docs/dev/table/connectors/upsert-kafka.zh.md:183]: `The messages are deserialized and serialized by formats`

**Comment by caozhen1937** [comment]
@fsk119 , Thank you for your comments, I fix some.

File: docs/dev/table/connectors/upsert-kafka.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fsk119) [reply docs/dev/table/connectors/upsert-kafka.md:50]: It's better to delete it. 
image

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 8]
```diff
@@ -29,36 +29,26 @@ under the License.
...
-key will fall into the same partition.
+作为 sink，upsert-kafka 连接器可以消费 changelog 流。它会将 INSERT/UPDATE_AFTER 数据作为正常的 Kafka 消息写入，并将 DELETE 数据以 value 为空的 Kafka 消息写入（表示对应 key 的消息被删除）。Flink 将根据主键列的值对数据进行分区，从而保证主键上的消息有序，因此同一 key 上的更新/删除消息将落在同一分区中。
 
-Dependencies
+依赖
 ------------
 
-In order to set up the upsert-kafka connector, the following table provide dependency information for
-both projects using a build automation tool (such as Maven or SBT) and SQL Client with SQL JAR bundles.
+为了设置 upsert-kafka 连接器，下表为项目使用自动化构建工具（例如 Maven 或 SBT ）以及 SQL 客户端使用 SQL JAR 包提供了依赖信息。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:41]: Please also delete it as you work in upsert-kafka.md. We should use `sql-connector-download-table.html` to generate the introduction. But currently, we only has English verion page here. 
image

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 9]
```diff
@@ -29,36 +29,26 @@ under the License.
...
-The Upsert Kafka connector allows for reading data from and writing data into Kafka topics in the upsert fashion.
+Upsert Kafka 连接器支持以 upsert 方式从 Kafka topic 中读取数据并将数据写入 Kafka topic。
 
-As a source, the upsert-kafka connector produces a changelog stream, where each data record represents
-an update or delete event. More precisely, the value in a data record is interpreted as an UPDATE of
-the last value for the same key, if any (if a corresponding key doesn’t exist yet, the update will
-be considered an INSERT). Using the table analogy, a data record in a changelog stream is interpreted
-as an UPSERT aka INSERT/UPDATE because any existing row with the same key is overwritten. Also, null
-values are interpreted in a special way: a record with a null value represents a “DELETE”.
+作为 source，upsert-kafka 连接器生产 changelog 流，其中每条数据记录代表一个更新或删除事件。更准确地说，数据记录中的 value 被解释为同一 key 的最后一个 value 的 UPDATE，如果有这个 key（如果不存在相应的 key，则该更新被视为 INSERT）。用表来类比，更改日志流中的数据记录被解释为 UPSERT，也称为 INSERT/UPDATE，因为任何具有相同 key 的现有行都被覆盖。另外，value 为空的消息将会被视作为 DELETE 消息。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:34]: 更改日志 -> changelog

File: docs/dev/table/connectors/upsert-kafka.zh.md [hunk 10]
```diff
@@ -29,36 +29,26 @@ under the License.
...
 
-As a sink, the upsert-kafka connector can consume a changelog stream. It will write INSERT/UPDATE_AFTER
-data as normal Kafka messages value, and write DELETE data as Kafka messages with null values
-(indicate tombstone for the key). Flink will guarantee the message ordering on the primary key by
-partition data on the values of the primary key columns, so the update/deletion messages on the same
-key will fall into the same partition.
+作为 sink，upsert-kafka 连接器可以消费 changelog 流。它会将 INSERT/UPDATE_AFTER 数据作为正常的 Kafka 消息写入，并将 DELETE 数据以 value 为空的 Kafka 消息写入（表示对应 key 的消息被删除）。Flink 将根据主键列的值对数据进行分区，从而保证主键上的消息有序，因此同一 key 上的更新/删除消息将落在同一分区中。
```
Comment (fsk119) [inline docs/dev/table/connectors/upsert-kafka.zh.md:36]: > Flink 将根据主键列的值对数据进行分区，从而保证主键上的消息有序，因此同一 key 上的更新/删除消息将落在同一分区中

Flink 将根据主键列的值对数据进行分区，从而保证主键上的消息有序，因此同一主键上的更新/删除消息将落在同一分区中。

感觉如果前面用中文的主键，那么后半句也用中文的描述会更为一致。

**Review by fsk119 — CHANGES_REQUESTED** [review]
LGTM except some minor problems.

File: docs/dev/table/connectors/upsert-kafka.zh.md (hunk 7 above)
Comment (fsk119) [reply docs/dev/table/connectors/upsert-kafka.zh.md]: oh. I don't notice the order in English verison. If you have time, please help me to adjust the order.

File: docs/dev/table/connectors/upsert-kafka.zh.md (hunk 10 above)
Comment (caozhen1937) [reply docs/dev/table/connectors/upsert-kafka.zh.md:36]: I agree with you.

**Comment by caozhen1937** [comment]
@fsk119 , Thanks for your comments, I fix it.

**Comment by fsk119** [comment]
LGTM. Thanks for your update.

**Review by fsk119 — APPROVED** [review]

**Comment by fsk119** [comment]
@wuchong CC

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request closed without merging**